lib/matplotlib/figure.py

```diff
@@ -2199,6 +2199,10 @@ def savefig(self, fname, *, transparent=None, **kwargs):
             :rc:`savefig.format` and the appropriate extension is appended to
             *fname*.
 
+            To write a binary format like PNG to ``sys.stdout``, use the
```

Comment:
This seems pretty in the weeds to me and adds a technical detail to an already long docstring.  I really think the user who wanted to write a png to standard out just didn't realize that a png is not a text format, because I can't see what the use is in dumping the contents to stdout.

Reply:
You could write a program that outputs the binary image and pipe the output to some other program, e.g. `python png_stdout.py | convert -trim - result.png`.

Maybe not the most common use case, but valid nevertheless.

Reply:
I'd say thats advanced enough usage that we can assume whoever may want to do that understands how to do the pipe on the python end.  I disagree that our docstring should document a python feature.

Reply:
There *is* a documentation problem on our side: We claim to support `file-like` objects. `sys.stdout` is one, but it does not work.

Alternatively we'd have to do something like:
~~~
fname : str or path-like or binary file-like
~~~
which is really terse and possibly not understandable unless you're an expert and already know about https://docs.python.org/3/library/sys.html#sys.stdout

Reply:
Sure, I won't block.  I just disagree that we need to specify what "file-like" means in the context of `stdout`.

Reply:
Will binary file-like work with svg/eps/etc?

Reply:
Yes, all formats will work with a binary file-like.